Implement basic functionality

walro opened the pull request:
I think it has what Bruce needs.

TODO:
- [X] Tests

walro committed:
Implement basic functionality

walro committed:
Fix README typo

walro commented:
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;CLRL3ORF&#x2F;1439-sprint34-bruce-problem-vid-omstarter-och-lokal-korning"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Sprint34: Bruce - Problem vid omstarter och lokal körning</a>

dentarg commented on lib/event_listener.rb:
```diff
@@ -0,0 +1,70 @@
+        @client.on_row_event_callback.call(row_event)
+      end
+    end
+  end
```
Let's do one file per class.

walro committed:
Split EventListener into different files

walro committed:
Move files to appropriate location

Mistake in last commit

walro committed:
Refactor callback setup

Let the event listeners handle their own callbacks instead of storing
them on the Client object

walro committed:
Remove #on_event on EventListerner

Confusing to set instance variables there to be used in subclasses.

twingly-mob committed:
Extract RowEvent to separate file

twingly-mob committed:
Refactor Client

Refactor and remove unused variables.

twingly-mob committed:
We only need write access to callback

twingly-mob committed:
Add specs for EventListeners
